src/psp-migration.md

```diff
@@ -0,0 +1,388 @@
+of the configuration they want to apply in their clusters. If they want to apply part of the PSP security checks it
+is not necessary to define the configurations related to the other fields.
+
+Once you have the Kubewarden instance running, it's time to deploy some policies to replace the `PodSecurityPolicy` object. Start by listing
+the PSPs in use. For the sake of this example, the following enforcements have been considered:
+
+- a PSP disabling privileged escalation
+- privileged containers
+- blocking pods running as root
+- forcing a particular user group
+- blocking host namespaces
+- allowing pod to use the port 443 only
+
+
+The yaml definition of the aforementioned PSP would look like the below:
+
+```console
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: restricted
+spec:
+  allowPrivilegeEscalation: false
+  runAsUser:
+    rule: MustRunAsNonRoot
+  supplementalGroups:
+    rule: MustRunAs
+    ranges:
+      - min: 1000
+        max: 65535
+  privileged: false
+  hostNetwork: false
+  hostIPC: false
+  hostPID: false
+  hostPorts:
+    - min: 443
+      max: 443
+```
+
+Let's create Kuberwarden policies to achieve the same goal. One thing that you need to know about Kubewarden policies is that every rule will be enforced by a separate policy. In our example, one policy each will be required for the user and group configuration, for the host namespace, for the privileged escalation and for the privileged container configuration.
+
+Let's start with blocking container escalation. For that you can deploy a policy as shown below:
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: ClusterAdmissionPolicy
+metadata:
+  name: psp-allowprivilegeescalation
+spec:
+  module: registry://ghcr.io/kubewarden/policies/allow-privilege-escalation-psp:v0.1.11
+  rules:
+    - apiGroups:
+        - ""
+      apiVersions:
+        - v1
+      resources:
+        - pods
+      operations:
+        - CREATE
+        - UPDATE
+  mutating: false
+  settings:
+    default_allow_privilege_escalation: false
+EOF
+```
+
+If you notice, we have specified `default_allow_privilege_escalation` to assume a value `false`. Once this policy starts running, it will restrict pods trying to run with more privileges than the parent container by default.
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: true
+  - name: sidecar
+    image: sidecar
+EOF
+Error from server: error when creating "STDIN": admission webhook "clusterwide-psp-allowprivilegeescalation.kubewarden.admission" denied the request: one of the containers has privilege escalation enabled
+```
+
+Now, to enforce the user and groups configuration, you can use the [user-group-psp](https://github.com/kubewarden/user-group-psp-policy) policy
```

Comment:
This is a nit because other policies have the word policy also hyperlinked. We can either delink them or link this one.
```suggestion
Now, to enforce the user and groups configuration, you can use the [user-group-psp policy](https://github.com/kubewarden/user-group-psp-policy)
```